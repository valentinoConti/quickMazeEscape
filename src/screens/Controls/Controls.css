#controls {
  width: 100%;
  height: 100%;
  text-align: center;
}

#controls > p {
  font-size: 36px;
  margin-top: 20px;
  margin-bottom: 20px;
}

#controls > p.title {
  margin-top: 30px;
  margin-bottom: 25px;
  font-size: 64px;
  font-weight: bold;
}

#controls > .menuButton {
  margin-top: 50px;
  width: 300px;
  height: 50px;
  margin-left: 0;
}

#objetive {
  display: flex;
  flex-direction: column;
}

#objetive > p {
  margin-top: 10px;
  margin-bottom: 10px;
  font-size: 20px;
}

#objetive > img {
  width: 60px;
  height: 60px;
  align-self: center;
}