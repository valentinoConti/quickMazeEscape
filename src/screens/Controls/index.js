import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import './Controls.css';

const Controls = ({setScreen}) => {
  const goMenu = () => { setScreen('menu') };

  return (
    <div id="controls">
      <p className="title">Controls:</p>
      <p>Arrow keys for moving</p>
      <p>"R" for restarting the game</p>
      <p className="title">Objetive:</p>
      <div id="objetive">
        <p>You should make the player</p>
        <img src={require('../../../assets/personaje.png')} />
        <p>get to the escape door</p>
        <img src={require('../../../assets/puerta.png')} />
        <p>as fast as posible and without getting out of movements!</p>
      </div>

      <button className="menuButton" onClick={goMenu}>Back to menu</button>
    </div>
  )
};

Controls.propTypes = {
  setScreen: PropTypes.func.isRequired
};

export default Controls;
