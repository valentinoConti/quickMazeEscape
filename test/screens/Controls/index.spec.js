import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import Controls from '../../../src/screens/Controls';

configure({adapter: new Adapter()});

describe('Controls screen test', () => {
  it('take snapshot and test button back to menu', () => {
    const mockSetScreen = jest.fn();
    const wrapper = mount(
      <Controls
        setScreen={mockSetScreen}
      />
    );

    expect(wrapper.find('#controls').exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
    expect(mockSetScreen).not.toHaveBeenCalled();
    wrapper.find('.menuButton').simulate('click');
    expect(mockSetScreen).toHaveBeenCalledWith('menu');
  });
});