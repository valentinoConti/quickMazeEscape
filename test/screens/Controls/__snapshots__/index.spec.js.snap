// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls screen test take snapshot and test button back to menu 1`] = `
<Controls
  setScreen={[MockFunction]}
>
  <div
    id="controls"
  >
    <p
      className="title"
    >
      Controls:
    </p>
    <p>
      Arrow keys for moving
    </p>
    <p>
      "R" for restarting the game
    </p>
    <p
      className="title"
    >
      Objetive:
    </p>
    <div
      id="objetive"
    >
      <p>
        You should make the player
      </p>
      <img
        src=""
      />
      <p>
        get to the escape door
      </p>
      <img
        src=""
      />
      <p>
        as fast as posible and without getting out of movements!
      </p>
    </div>
    <button
      className="menuButton"
      onClick={[Function]}
    >
      Back to menu
    </button>
  </div>
</Controls>
`;
